common/ferrostar/src/navigation_controller/mod.rs

```diff
@@ -139,6 +144,11 @@ impl NavigationController {
                             *snapped_user_location,
                             &self.route.get_linestring(),
                         );
+                        let snapped_user_location_with_course = apply_snapped_course(
```

Comment:
I was using this in the trip state constructed just below it. But reverted that to see if the memory heap stall went away.

Reply:
I believe the memory heap issue was due to the incorrect bounds checking logic I flagged above.